form.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+    function func(String $text, int $size, &$valid, &$errors){
+        $tmp = $_POST["$text"];
+        if(!$tmp){
+            $errors["$text"] = "Полето е задължително."; 
+        } elseif(strlen($tmp) > $size ){
+            $errors["$text"] = "Полето има максимална дължина $size символа.";
+        } else {
+            $valid[$text] = $tmp;
+        }        
+    }
+
+    $valid = array();
+    $errors = array();
+    
+    if( $_POST ){
+        func("title", 150, $valid, $errors);
+        func("teacher",200, $valid, $errors);
+
+        $description = $_POST['description'];
+        if(!$description){
+            $errors['description'] = "Описанието е задължително."; 
+        } elseif(strlen($description) < 10 ){
+            $errors['description'] = "Описанието има минимална дължина 10 символа.";
+        } else {
+            $valid['description'] = $description;
+        }
+
+        $group = $_POST['group'];
+        if(!$group){
+            $errors['group'] = "Не сте избрали група.";
+        } else {
+            $valid['group'] = $group;
+        }
+
+        $credits = $_POST['credits'];
+        if (!$credits){
+            $errors['credits'] = "Не сре задали кредити.";
+        } elseif($credits < 0 || $credits > 20){
+            $errors['credits'] = "Кредити могат да са от 0 до 20.";
+        } else {
+            $valid['credits'] = $credits;
+        }
+
+        if(count($valid) == 5 ){
+
+            $filename = 'results.txt';
```

Comment:
When I submit the form, I receive no result on the screen. Was everything successful? Did something fail? What exactly failed? What do I need to do to fix it? How can I (as a dumb user not knowing about the "back" button of the browser) navigate to the previous page and try to fill the form once again? What was the exact information that was recorded by the server to the file?

Reply:
Here is the acceptance criteria for the task: https://iliasky.com/www/presentations/Forms.html?read=1#11